Tests/SonarScanner.MSBuild.PreProcessor.Test/ArgumentProcessorTests.cs

```diff
@@ -52,19 +52,27 @@ public void PreArgProc_MissingArguments()
             // 3. Only key and host URL are required
             var args = CheckProcessingSucceeds("/k:key", "/d:sonar.host.url=myurl");
             "key".Should().Be(args.ProjectKey);
-            "myurl".Should().Be(args.SonarQubeUrl);
+            args.SonarServer.Should().BeOfType<SonarQubeServer>().Which.ServerUrl.Should().Be("myurl");
 
             // 4. Argument is present but has no value
             logger = CheckProcessingFails("/key:");
             logger.AssertSingleErrorExists("/key:");
             logger.AssertErrorsLogged(1);
         }
 
+        [TestMethod]
+        public void PreArgProc_HostAndSonarcloudUrlError()
+        {
+            var logger = CheckProcessingFails("/k:key", "/d:sonar.host.url=firstUrl", "/d:sonar.scanner.sonarcloudUrl=secondUrl");
+            logger.AssertErrorLogged("The sonar.host.url value and sonar.scanner.sonarcloudUrl are both set and are different. " +
+                "Set either sonar.host.url for SonarQube and sonar.scanner.sonarcloudUrl for SonarCloud.");
+        }
+
         [TestMethod]
         public void PreArgProc_DefaultHostUrl()
         {
             var args = CheckProcessingSucceeds("/k:key");
-            "http://localhost:9000".Should().Be(args.SonarQubeUrl);
+            args.SonarServer.Should().BeOfType<SonarCloudServer>().Which.ServerUrl.Should().Be("https://sonarcloud.io");
```

Comment:
This is the breaking change:
`http://localhost:9000` -> `https://sonarcloud.io` as defined in [Scanner Bootstrappers implementation guidelines](https://xtranet-sonarsource.atlassian.net/wiki/spaces/LANG/pages/3155001395/Scanner+Bootstrappers+implementation+guidelines)